lock-api/src/main/java/com/palantir/lock/client/LockRefreshingTimelockService.java

```diff
@@ -39,19 +41,47 @@
 
     private final TimelockService delegate;
     private final LockRefresher lockRefresher;
-
+    private final Optional<ScheduledExecutorService> managedExecutor;
+
+    /**
+     * @deprecated Use {@link #create(TimelockService, ScheduledExecutorService)} instead.
+     *
+     * @param timelockService The {@link TimelockService} to wrap
+     * @return A {@link TimelockService} that automatically refreshes locks
+     */
+    @Deprecated
     public static LockRefreshingTimelockService createDefault(TimelockService timelockService) {
         ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor(new ThreadFactoryBuilder()
                 .setNameFormat(LockRefreshingTimelockService.class.getSimpleName() + "-%d")
                 .setDaemon(true)
                 .build());
         LockRefresher lockRefresher = new LockRefresher(executor, timelockService, REFRESH_INTERVAL_MILLIS);
-        return new LockRefreshingTimelockService(timelockService, lockRefresher);
+        return new LockRefreshingTimelockService(timelockService, lockRefresher, executor);
+    }
+
+    /**
+     * Creates a {@link TimelockService} that uses the specified executor to regularly refresh
+     * locks that have been taken out.
+     *
+     * @param timelockService The {@link TimelockService} proxy to wrap
+     * @param executor An executor service whose lifecycle is managed by the consumer of this method
+     * @return The {@link TimelockService} that automatically refreshes locks
+     */
+    public static LockRefreshingTimelockService create(
+            TimelockService timelockService,
+            ScheduledExecutorService executor) {
+        LockRefresher lockRefresher = new LockRefresher(executor, timelockService, REFRESH_INTERVAL_MILLIS);
+        return new LockRefreshingTimelockService(timelockService, lockRefresher, null);
     }
 
-    public LockRefreshingTimelockService(TimelockService delegate, LockRefresher lockRefresher) {
+    @VisibleForTesting
+    LockRefreshingTimelockService(
```

Comment:
this is technically an API break as it's no longer public, but I think that's fine?